text/0032-dust-protection/0032-dust-protection.md

```diff
@@ -0,0 +1,114 @@
+                                        </td>
+                                    </tr>
+                                    <tr>
+                                        <td valign="top">Address </td>
+                                        <td colspan="2">
+                                                                     <details>
+                                                <summary>Ed25519 Address</summary>
+                                                <table>
+                                                    <tr>
+                                                        <td><b>Name</b></td>
+                                                        <td><b>Type</b></td>
+                                                        <td><b>Description</b></td>
+                                                    </tr>
+                                                    <tr>
+                                                        <td>Address Type</td>
+                                                        <td>uint8</td>
+                                                        <td>
+                                                            Set to <strong>value 1</strong> to denote an <i>Ed25519 Address</i>.
+                                                        </td>
+                                                    </tr>
+                                                    <tr>
+                                                        <td>Address</td>
+                                                        <td>Array&lt;byte&gt;[32]</td>
+                                                        <td>The raw bytes of the Ed25519 address which is a BLAKE2b-256 hash of the Ed25519 public key.</td>
+                                                    </tr>
+                                                </table>
+                                            </details>
+                                        </td>
+                                    </tr>
+                                    <tr>
+                                        <td>Amount</td>
+                                        <td>uint64</td>
+                                        <td>The amount of tokens to deposit with this <i>SigLockedSingleOutput</i> output.</td>
+                                    </tr>
+                                </table>
+
+
+### The new protocol dust rule
+
+Let `A` be the address that should hold the dust outputs' balances. Let `M` be the sum of all SigLockedDustAllowanceOutputs' values on `A`.So the total allowed dust outputs on `A` is (`M`/1 MIOTA) * 100 rounded down to the nearest integer. So 100 outputs for 1 MIOTA deposited.
+
+The SigLockedDustAllowanceOutput must be greater or equal to 1MI.
+
+If while processing a milestone a new dust UTXO is created but there is not a sufficient amount already locked in SigLockedDustAllowanceOutputs then the encompassing message should be marked as `ignored`.
```

Comment:
This introduces a race condition: an attacker could always create a new dust UTXO between milestones epochs for the unlocking of the deposit UTXO to be ignored. While this will perhaps not ever be a money issue (1MIOTA is a negligible amount of money) it may represent a processing issue: some use-case may rely on this deposit and address to be invalidated for good. Therefore, I think we need to define better how dust consolidation and deposit unlock look like.

Reply:
In our discussions we considered this not a serious issue

We assumed that the receiver can always consolidate the attacker's outputs and just get free money as long as the attack continues. However, your point have to be validated with bizdev probably

Reply:
Yes, there is some race-condition/censoring where an attacker always tries to issue a dust transaction before (wrt to the WhiteFlag order) the spend transaction of a `SigLockedDustAllowanceOutputs`. I guess this is exactly, what you guys are talking about?
I think this is an important issue and we should talk about it in the RFC. This has been one of my main concerns with the dust protection without a special output type. But I think it is much less important in this scenario: First, you can just ignore SigLockedDustAllowanceOutputs, and just move everything else without any possibilities for race-conditions (this should be sufficient for the usual "wallet" use-case). Secondly, when you are under attack and really need to spend everything, you can still "dust yourself" until the limit is reached and then issue one big transaction consuming everything at once.

Reply:
> First, you can just ignore SigLockedDustAllowanceOutputs, and just move everything else without any possibilities for race-conditions (this should be sufficient for the usual "wallet" use-case).

I don't get this first point. Are you suggesting a special WF "ordering" of dust-related transactions?

Reply:
I mean, as an honest user whose address got polluted by an attacker sending a SigLockedDustAllowanceOutputs and tons of dust outputs, you can just ignore all the dust and only transfer away regular SigLockedOutputs of >= 1 Mi. When doing this there cannot be any race condition and it would be the exact same result as when an honest users could "block" SigLockedDustAllowanceOutputs.

This is true without any changes to the validation or WF ordering.

Reply:
That scenario is clear: as an attacker, you cannot deny a user to spend their regular funds. What I was referring to is that, as an attacker, you can deny a user to spend its last `SigLockedDustAllowanceOutput`.

Reply:
This md discusses the issue at length and offers mitigations:
https://hackmd.io/@iota-protocol/S1zFaBCjP

Reply:
I told wollac yesterday about a special "lock type" to stop collecting dust. This lock type needs to be signed with the private key for the address. This way you could stop collecting dust in the first milestone, and spend your funds without a race condition in the next milestone.
We came to the conclusion that we won't add this lock type now, but I just wanted to write it down, so the idea is not lost.

Reply:
Why did you come to the conclusion that this shouldn't be added?
Doesn't sound too complicated